test/parallel/test-net-connect-write-queue-size.js

```diff
@@ -0,0 +1,31 @@
+'use strict';
+const common = require('../common');
+const assert = require('assert');
+const net = require('net');
+const tmpdir = require('../common/tmpdir');
+
+tmpdir.refresh();
+
+const socket = new net.Socket();
+assert.strictEqual(socket.writeQueueSize, -1);
+
+const server = net.createServer()
+  .listen(common.PIPE, common.mustCall(() => {
+    // The pipe connection is a synchronous operation
+    // `net.connect` will set `socket.connecting` to true
+    const socket = net.connect(common.PIPE, common.mustCall(() => {
+      socket.destroy();
+      server.close();
+    }));
+    // Set connecting to false here to make `socket.write` can call into
+    // libuv directly (see `_writeGeneric` in `net.js`).
+    // because the socket is connecting currently in libuv, so libuv will
+    // insert the write request into write queue, then we can get the
+    // size of write queue by `socket.writeQueueSize`.
+    socket.connecting = false;
```

Comment:
Is there a way to test the feature without this hack? I mean, how would `socket.writeQueueSize` will be used in the real world? The tests should simulate that.

Reply:
Yes, it is difficult to write the test, and i think this test is not good.`writeQueueSize` means how many bytes are waiting to write in libuv. and add `socket._handle.writeQueueSize` to the count of `bytesWritten` maybe a good idea. I'll take a look later, thanks !